frontend/src/pages/TermPlanner/TermPlanner.tsx

```diff
@@ -34,11 +36,40 @@ const TermPlanner = () => {
   const planner = useSelector((state: RootState) => state.planner);
   const degree = useSelector((state: RootState) => state.degree);
 
-  const [termsOffered, setTermsOffered] = useState<Term[]>([]);
-  const [isDragging, setIsDragging] = useState(false);
+  const [draggingCourse, setDraggingCourse] = useState<string | undefined>();
 
   const dispatch = useDispatch();
 
+  useEffect(() => {
+    // update the legacy term offered
+    const updateOfferingsData = async () => {
+      const years: string[] = [];
+      for (let i = 0; i < planner.numYears; i++) {
+        years.push((planner.startYear + i).toString());
+      }
+
+      // get the outdated legacy offerings
+      const outdated = Object.entries(planner.courses)
+        .filter(([, course]) => {
+          const oldOfferings = course.legacyOfferings;
+          // no legacy offering data, or doesn't include the wanted years
+          return !oldOfferings || years.filter((y) => !(y in oldOfferings)).length > 0;
+        })
+        .map(([code]) => code);
+
+      // update them
+      const data = await getAllCourseOfferings(outdated, years);
+      Object.entries(data).forEach(([code, offerings]) => {
+        dispatch(updateLegacyOfferings({ code, offerings }));
+      });
+    };
+
+    updateOfferingsData();
+    // hacky solution to useEffect deps, because it cant compare arrays well enough :c
+    // also only want it to update on new/removed courses, not altered inner state
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Is it possible to avoid disabling this eslint error? If it triggers useEffect often, try using useCallback/useMemo. Lmk if it isnt